packages/core/src/extension/extension-messenger.ts

```diff
@@ -0,0 +1,96 @@
+import { IframeEventListener, IframeEventPoster } from '../common/iframe';
+
+import { ExtensionNotFoundError } from './errors';
+import { Extension } from './extension';
+import { ExtensionCommandHandlers } from './extension-command-handler';
+import { ExtensionOriginEventMap, stringToExtensionCommand } from './extension-origin-event';
+import { HostOriginEvent } from './host-origin-event';
+
+export class ExtensionMessenger {
+    private _extensions: Extension[] | undefined;
+    private _listeners: { [extensionId: string]: IframeEventListener<ExtensionOriginEventMap> } =
+        {};
+    private _posters: { [extensionId: string]: IframeEventPoster<HostOriginEvent> } = {};
+
+    listen(
+        extensions: Extension[],
```

Comment:
I'm thinking you can just inject `ReadableCheckoutStore`  as a constructor parameter of `ExtensionMessenger`. That way you don't need to pass `extensions`. You can just call the following to get the available extensions.

```ts
const { data: { getExtensions } } = this.store.getState();
const extensions = getExtensions();
```

Reply:
Sure. Now the constructor is:
```
constructor(
        private _store: CheckoutStore,
        private _listeners: { [extensionId: string]: IframeEventListener<ExtensionOriginEventMap> },
        private _posters: { [extensionId: string]: IframeEventPoster<HostOriginEvent> },
    ) {}
```